internal/users/users.go

```diff
@@ -161,6 +165,116 @@ func computeGroupOperation(newGroupsInfo []GroupInfo, currentLocalGroups []strin
 	return groupsToAdd, groupsToRemove
 }
 
+// CleanUserFromSystemGroups removes the user from all system groups.
+func CleanUserFromSystemGroups(user string, args ...Option) (err error) {
+	defer decorate.OnError(&err, "could not clean user %q from system groups", user)
+
+	opts := defaultOptions
+	for _, arg := range args {
+		arg(&opts)
+	}
+
+	// Get the list of groups the user belong to
+	groups, err := existingLocalGroups(user, opts.groupPath)
+	if err != nil {
+		return err
+	}
+
+	for _, group := range groups {
+		args := opts.gpasswdCmd[1:]
+		args = append(args, "--delete", user, group)
+		if err := runGPasswd(opts.gpasswdCmd[0], args...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// CleanSystemGroups removes all unexistent users from the system groups.
+//
+// In order to do that, it needs to query NSS for all existent users and then look at /etc/groups.
+func CleanSystemGroups(args ...Option) (err error) {
+	defer decorate.OnError(&err, "could not clean system groups completely")
+
+	opts := defaultOptions
+	for _, arg := range args {
+		arg(&opts)
+	}
+
+	// Get the list of users from NSS
+	// #nosec:G204 - the cmd arguments are only overridable in a controlled way during tests.
+	cmd := exec.Command(opts.getUsersCmd[0], opts.getUsersCmd[1:]...)
```

Comment:
Have you considered using something like https://github.com/linuxdeepin/go-lib/blob/master/users/passwd/passwd.go#L86C1-L89C1 instead of subprocessing, by doing the cgo calls directly? It seems indeed that having that built-in in the standard library has been declined (https://github.com/golang/go/issues/47907).

Reply:
Won't this affect our control over the output of the command (and make it harder for us to mock it)? 
I'm never sure if sub-processing is better than `CGo`ing (or vice-versa). Usually, I avoid CGo at all costs, but maybe we can talk more about it and decide what's the best approach?

Reply:
I still thing that cgoing is better than subprocess when we can. I think you can do the same: host on the type an interface  with one call (or just a function actually). Then, the real implementation will do the cgo call (only copy the needed part) and will be tested as unit test (as we can’t mock its content, like ensuring on a real system that it returns a non empty list and at least root).

Then, for testing, we pass that function as functional argument instead of the mock subprocess, and the rest of the logic is similar to what you already have in the subprocess mock (failing case with special keyword and returning an hardcoded list)